students/hyska_monika/lesson_10_organizing_files/filling_in_the_gaps.py

```diff
@@ -0,0 +1,35 @@
+"""
+Program finds all files with a given prefix, such as spam001.txt,
+spam002.txt, and so on, in a single folder and locates any gaps
+in the numbering (such as if there is a spam001.txt and spam003.txt but no
+spam002.txt). Have the program rename all the later files to close this gap.
+"""
+import shutil
+import os
+import re
+
+
+def search_prefix_add_gaps(path, prefix):
+    number = 1
+    for file_name in os.listdir(path):
+        file_with_prefix = prefix.search(file_name)
```

Comment:
This will also catch names like:
`spam000.txtwhateverspam001.txt` and `notspam001.txttxt`
If you want a prefix, then it should be `match` and if you don't want extra endings, then you need a dollar sign at the end.